Makefile

```diff
@@ -246,7 +246,8 @@ format-check:
 HELP_onnx_generator=generate various onnx sources and headers
 TARGET+=onnx_generator
 onnx_generator:
-	pip install -r requirements.txt
+	python -m venv venv
+	. venv/bin/activate; pip install -r requirements.txt
```

Comment:
would make the venv as a standalone target in preparation for other python related stuff and to not run pip each time

Reply:
I like having it here so we are 100% sure that the correct version is used. Using `venv` does not remove or override the existing virtual environment (if any), so I don't think its a problem. If the environment already exists it will continue and if the version is already installed, nothing will happen.

Reply:
but we have no way of modifying the venv prior the execution.
let's say we want to test something and install another onnx version inside the venv, it will get overwritten without mercy.
I see the `requirements.txt` more as a setup configuration, not as a execution configuration.
no big issue on my side though

Reply:
Will close this. If in the future we need to install other versions, we can discuss it further.